File: infrastruktur/src/main/java/com/github/haschi/haushaltsbuch/infrastruktur/ResultBuilder.java
package com.github.haschi.haushaltsbuch.infrastruktur;

import io.vertx.ext.web.RoutingContext;
import org.github.haschi.haushaltsbuch.infrastruktur.modellierung.de.Aggregatkennung;

import java.util.function.Consumer;

public class ResultBuilder
{
    Consumer<RoutingContext> location(final String url, final Aggregatkennung id) {
        return (RoutingContext context) -> {
            context.response()
                    .putHeader("Location", url)
                    .setStatusCode(200)
                    .end();
        };
    }

    public Aggregatkennung id()
    {
        return null;
    }
}

File: infrastruktur/src/main/kotlin/com/github/haschi/haushaltsbuch/infrastruktur/ResultBuilder.kt
package com.github.haschi.haushaltsbuch.infrastruktur

import io.vertx.ext.web.RoutingContext
import com.github.haschi.domain.haushaltsbuch.modell.core.values.Aggregatkennung


class ResultBuilder
{
    internal fun location(url: String, id:Aggregatkennung): (RoutingContext) -> Unit
    {
        return { context: RoutingContext ->
            context.response()
                    .putHeader("Location", url)
                    .setStatusCode(200)
                    .end()
        }
    }

    fun id(): Aggregatkennung?
    {
        return null
    }
}
